packages/webpack/src/migrations/update-19-6-3/proxy-config.ts

```diff
@@ -0,0 +1,55 @@
+import { logger, Tree, updateJson } from '@nx/devkit';
+import { forEachExecutorOptions } from '@nx/devkit/src/generators/executor-options-utils';
+import { WebDevServerOptions } from '../../executors/dev-server/schema';
+
+export default async function update(tree: Tree): Promise<void> {
+  const migrate = (options: WebDevServerOptions) => {
+    if (!options.proxyConfig) return;
+    if (options.proxyConfig.endsWith('.json')) {
+      updateJson(tree, options.proxyConfig, (json) => {
+        if (Array.isArray(json)) return json;
+        if (typeof json === 'object') {
+          return Object.keys(json).map((context) => ({
+            context: [context],
+            ...json[context],
+          }));
+        }
+        return json;
+      });
+    } else {
+      // For non-JSON files, it's not possible to automatically update the proxy config
+      // since its content can vary greatly.
+      logger.warn(`${options.proxyConfig} needs to be updated manually.
```

Comment:
This could be noisy if logged for multiple proxy config files. It might be better to aggregate the proxy config file paths and log the warning once at the end, listing them.